### PR TITLE
Fixed Bugs that prevented results to be uploaded to Xray Cloud

### PR DESCRIPTION
**Pull request by qytera-development**
Fixed Bugs that prevented results to be uploaded to Xray Cloud

### CODE REVIEW

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java [hunk 1]
```diff
@@ -63,9 +63,29 @@ public XrayTestEntity buildTestEntity(XrayTest xrayTest, T scenarioData) {
...
+        if (scenarioData instanceof TestScenarioLogCollection) {
+            entity.setSteps(getSteps(xrayTest, scenarioData));
+        } else {
+            entity.setSteps(null);
+        }
+
+        // Empty arrays for examples are not accepted by the Xray API
+        List<String> examples = getExamples(xrayTest, scenarioData);
+        if (examples != null && examples.size() > 0) {
+            entity.setExamples(getExamples(xrayTest, scenarioData));
```
Comment (csvtuda) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java:77]: This could be simplified to:

```java
entity.setExamples(examples);
```

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/config/XrayConfigHelper.java [hunk 2]
```diff
@@ -63,6 +63,18 @@ public class XrayConfigHelper {
...
+     */
+    public static final String RESULTS_UPLOAD_ENVIRONMENTS_ENABLED = "xray.resultsUpload.environments.enabled";
+    /**
+     * The selector for accessing the configured assignee to assign text execution issues to.
+     */
+    public static final String RESULTS_UPLOAD_ENVIRONMENTS_OS = "xray.resultsUpload.environments.os";
+    /**
+     * The selector for accessing the configured assignee to assign text execution issues to.
+     */
+    public static final String RESULTS_UPLOAD_ENVIRONMENTS_DRIVER = "xray.resultsUpload.environments.driver";
```
Comment (csvtuda) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/config/XrayConfigHelper.java:77]: The Javadoc comments do not match the configuration keys, I think.

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java [hunk 3]
```diff
@@ -63,9 +63,29 @@ public XrayTestEntity buildTestEntity(XrayTest xrayTest, T scenarioData) {
...
+        // Only when there is one execution we can add steps directly to the test entity
+        if (scenarioData instanceof TestScenarioLogCollection) {
+            entity.setSteps(getSteps(xrayTest, scenarioData));
+        } else {
+            entity.setSteps(null);
+        }
+
+        // Empty arrays for examples are not accepted by the Xray API
+        List<String> examples = getExamples(xrayTest, scenarioData);
+        if (examples != null && examples.size() > 0) {
```
Comment (csvtuda) [inline qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java:76]: `!examples.empty()` is probably cleaner code.

**Review by csvtuda — APPROVED** [review]
I think we probably should have more test in the future for empty `examples` etc., but it doesn't matter for now 👍

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java (hunk 1 above)
Comment (qytera-development) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java:77]: Ok, fixed it

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java (hunk 3 above)
Comment (qytera-development) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/builder/test/XrayTestEntityBuilder.java:76]: Ok, fixed it

File: qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/config/XrayConfigHelper.java (hunk 2 above)
Comment (qytera-development) [reply qtaf-xray-plugin/src/main/java/de/qytera/qtaf/xray/config/XrayConfigHelper.java:77]: Ok, fixed it
